pkg/kvdpa/device.go

```diff
@@ -254,7 +254,27 @@ func GetVdpaDevicesByPciAddress(pciAddress string) ([]VdpaDevice, error) {
 		return nil, unix.EINVAL
 	}
 
-	return GetVdpaDevicesByMgmtDev(busName, mgmtDeviceName)
+	msgs, err := GetNetlinkOps().RunVdpaNetlinkCmd(VdpaCmdDevGet, syscall.NLM_F_DUMP, nil)
```

Comment:
You are not "making a more selective request down to netlink." This is the same netlink command that is used in `ListVdpaDevices`

What's the difference between this and using `GetVdpaDevicesByMgmtDev`? Only that `ListVdpaDevices()` calls `getBusInfo()`?

If so, a more generic way to solving this without duplicating 90% of the code would be to  add an argument to `ListVdpaDevices` with a bus name and mgtdev name to filter (that can be empty) and forward those arguments down to `parseDevLinkVdpaDevList` who would now do filtering before filling up businformation. Would that work for you?

Reply:
Removed unneeded public functions `GetVdpaDevicesByMgmtDev` and `GetVdpaDevicesByPciAddress`. `ListVdpaDevices` is the good one